src/components/dropdown-filter-ajax/dropdown-filter-ajax.stories.js

```diff
@@ -0,0 +1,138 @@
+
+const store = new Store({
+  value: ''
+});
+
+const create = (evt, component) => component.state.filter;
+
+// Shared Props
+const onChange = (evt) => {
+  store.set({ value: evt.target.value });
+  action('change')();
+};
+const autoFocus = boolean('autoFocus', false);
+const cacheVisibleValue = boolean('cacheVisibleValue', true);
+const disabled = boolean('disabled', false);
+const name = text('name', 'Name');
+const readOnly = boolean('readOnly', false);
+const timeToDisappear = number('timeToDisappear', 0);
+const label = text('label', 'Dropdown Label');
+const labelInline = boolean('labelInline', true);
+const labelWidth = labelInline ? text('labelWidth', '') : null;
+const labelAlign = labelInline ? select('labelAlign', OptionsHelper.alignBinary) : null;
+const labelHelp = text('labelHelp', 'This is help text');
+const inputWidth = text('inputWidth', '');
+const fieldHelp = text('fieldHelp', 'This is field help text');
+const fieldHelpInline = boolean('fieldHelpInline', false);
+const createText = text('createText', '');
+const createIconType = text('createIconType', '');
+const suggest = boolean('suggest', false);
+const freetext = boolean('freetext', false);
+const path = text('path', '/');
+const acceptHeader = text('acceptHeader', 'application/json');
+const rowsPerRequest = text('rowsPerRequest', '25');
+const dataRequestTimeout = number('dataRequestTimeout', 500);
+const withCredentials = boolean('withCredentials', false);
+const options = ImmutableHelper.parseJSON([
+  {
+    id: 1, name: 'Orange'
+  }, {
+    id: 2, name: 'Blue'
+  }
+]);
+
+storiesOf('DropdownFilterAjax', module)
+  .addParameters({
+    info: {
+      propTablesExclude: [State]
+    }
+  })
+  .add('default', () => {
+    return (
+      <State store={ store }>
+        <DropdownFilterAjax
+          autoFocus={ autoFocus }
+          cacheVisibleValue={ cacheVisibleValue }
+          disabled={ disabled }
+          name={ name }
+          readOnly={ readOnly }
+          timeToDisappear={ timeToDisappear }
+          label={ label }
+          labelInline={ labelInline }
+          labelWidth={ labelWidth }
+          labelAlign={ labelAlign }
+          labelHelp={ labelHelp }
+          inputWidth={ inputWidth }
+          fieldHelp={ fieldHelp }
+          fieldHelpInline={ fieldHelpInline }
+          createText={ createText }
+          createIconType={ createIconType }
+          suggest={ suggest }
+          freetext={ freetext }
+          path={ path }
+          acceptHeader={ acceptHeader }
+          rowsPerRequest={ rowsPerRequest }
+          getCustomHeaders={ () => true }
+          dataRequestTimeout={ dataRequestTimeout }
+          withCredentials={ withCredentials }
+          options={ options }
+          onChange={ onChange }
+          value={ store.get('value') }
+        />
+      </State>
+    );
+  }, {
+    notes: { markdown: notes }
+  })
+  .add('withCreate', () => {
+    return (
+      <State store={ store }>
```

Comment:
Have you tried it if changing state in the first story doesn't have impact on the second story?

Reply:
The state is shared, which to my mind, is perfectly fine, so long as the properties which make the stories different are not shared.

Again, interested to hear other opinions.